scripts/sandbox.js

```diff
@@ -1,8 +1,11 @@
-let sandboxLevel = "";
-let sandboxObject = ObstacleCube;
-let sanboxTpX = 3;
-let sandboxTpY = 3;
-let sandboxPortalColor = "Blue";
+const sandbox = {
+  changes: [""],
```

Comment:
How about making this array empty? The first item don't seem valid anyways.

Reply:
Yeah, I just didn't think it would work because of a Numbers of reasons. And I just wanted it to gosh darn work:)

Reply:
This won't work with our current ways of doing things. Since we want to be able to return to nothing being shown this isn't possible. This change would mean that the "first" change would be that one object is added. That means that it will always be shown and we cannot go to the absolute beginning. I tried this and this is what it says if I log the `sandbox.history`:

`["CubeCreator.create({ ↵      type: ObstacleCube, ↵ …lue",↵      tpX: 0,↵      tpY: 0↵    }).update();"]`

There is only one value in the array, and we cannot splice to remove it anymore because then it won't work to say: `eval(sandbox.level)` since the value of `sandbox.level` will be undefined or something else weird.

Reply:
If we instead would have the empty string as our first value we would (if we pressed undo enough times) come to an empty string which would mean that nothing would have shown.
Like this:

Before undo:
`["", "CubeCreator.create({ ↵      type: ObstacleCube, ↵ …lue",↵      tpX: 0,↵      tpY: 0↵    }).update();"]`

After:

`[""]`

Reply:
And about it not being valid: It is, it just says that it should not show anything. We could instead use other methods, ofcourse, like instead having the value as `null` or `undefined` but that would be more characters than us just using an empty string...

Reply:
I'd still save it as an empty array and done something like.

```javascript
const level = history[index] || "";
eval(level);
```

Or even:
```javascript
const level = history[index];

if (level) {
 eval(level)
}
```

I found a bug regarding this btw. :)

Reproduction steps:
1. Place a cube.
2. Press undo.
3. Place a cube.

Expected: A cube is placed.
Actual: An error is thrown saying "VM54813:1 Uncaught ReferenceError: nullCubeCreator is not defined"

Reply:
You mean that your solution throws an error? 'Cause I didn't get that problem?